pkgs/tools/compression/zopfli/default.nix

```diff
@@ -25,6 +25,11 @@ stdenv.mkDerivation rec {
     })
   ];
 
+  # zopfli hard-codes gcc/g++, but we want to build with clang on MacOS
+  postPatch = stdenv.lib.optionalString stdenv.isDarwin  ''
+    substituteInPlace Makefile --replace gcc cc --replace g++ c++ --replace -soname -install_name
+  '';
```

Comment:
See https://github.com/NixOS/nixpkgs/issues/32171#issuecomment-347895060.

Linux has `clangStdenv`, and currently package won't build with it, only with default `stdenv`. It would be better if you fix this for all platforms, and not just Darwin:

```
  postPatch = ''
    substituteInPlace Makefile --replace gcc cc --replace g++ c++
  '' + stdenv.lib.optionalString stdenv.isDarwin ''
    substituteInPlace Makefile --replace -soname -install_name
  '';
```

Reply:
This is not the only problem with this Makefile. install_name should be absolute, there should be no `.so` on Darwin. I'll remake this as a patch.

Reply:
I generally leave the extension alone if upstream doesn't handle it correctly, since it doesn't matter.

Reply:
Ok, let me know if I should do any changes (e.g. remove the comment etc)